eng/common/performance/performance-setup.ps1

```diff
@@ -0,0 +1,100 @@
+Param(
+    [string] $SourceDirectory=$env:BUILD_SOURCESDIRECTORY,
+    [string] $CoreRootDirectory,
+    [string] $Architecture="x64",
+    [string] $Framework="netcoreapp3.0",
+    [string] $CompilationMode="Tiered",
+    [string] $Repository=$env:BUILD_REPOSITORY_NAME,
+    [string] $Branch=$env:BUILD_SOURCEBRANCH,
+    [string] $CommitSha=$env:BUILD_SOURCEVERSION,
+    [string] $BuildNumber=$env:BUILD_BUILDNUMBER,
+    [string] $RunCategories="coreclr corefx",
+    [string] $Csproj="src\benchmarks\micro\MicroBenchmarks.csproj",
+    [switch] $Internal,
+    [string] $Configurations
+)
+
+. $PSScriptRoot\..\pipeline-logging-functions.ps1
+
+if ($Repository -eq "dotnet/performance") {
+    $RunFromPerformanceRepo = $true;
+}
+else {
+    $RunFromPerformanceRepo = $false;
+}
+
+if ($Configurations -eq [string]::Empty) {
+    $Configurations="CompilationMode=$CompilationMode"
+}
+
+if ($CoreRootDirectory -eq [string]::Empty) {
+    $UseCoreRun = $false
+}
+else {
+    $UseCoreRun = $true
+}
+
+$PayloadDirectory = (Join-Path $SourceDirectory "Payload")
+$PerformanceDirectory = (Join-Path $PayloadDirectory "performance")
+$WorkItemDirectory = (Join-Path $SourceDirectory "workitem")
+$Creator = ""
+
+if ($Internal) {
+    $Queue = "Windows.10.Amd64.ClientRS1.Perf"
+    $PerfLabArguments = "--upload-to-perflab-container"
+    $ExtraBenchmarkDotNetArguments = ""
+    $Creator = ""
+    $HelixSourcePrefix = "official"
+}
+else {
+    if ($Framework.StartsWith("netcoreapp")) {
```

Comment:
In this else block, we probably could have something that checks if this is functional or perf testing.